### PR TITLE
Add etw UnicodeStringField

### PR DESCRIPTION
**Pull request by pavelbazika**
Allow to send UTF-16 encoded message. It's the only missing part needed for writing messages with local characters into event log via this library, when string is specified as win:UnicodeString in instrumentation manifest (which cannot be changed).

`UnicodeStringArray` is there only for completeness from my point of view.

### CODE REVIEW

**Review by anmaxvl — APPROVED** [review]
LGTM. can you commit with `-s`?

**Review by dcantah — APPROVED** [review]
lgtm also, same ask as above though. Can do `git commit --amend --sign-off` and repush.

File: pkg/etw/fieldopt.go [hunk 1]
```diff
@@ -66,6 +74,17 @@ func StringArray(name string, values []string) FieldOpt {
 	}
 }
 
+// UnicodeStringArray adds an array of UTF-16 strings to the event.
+func UnicodeStringArray(name string, values []string) FieldOpt {
+	return func(em *eventMetadata, ed *eventData) {
+		em.writeArray(name, inTypeUnicodeString, outTypeString, 0)
```
Comment (kevpar) [inline pkg/etw/fieldopt.go:80]: Do we want the out type here? It looks like `TraceLoggingWideString` in `TraceLoggingProvider.h` just uses `TlgInUNICODESTRING` with no out type.

File: pkg/etw/fieldopt.go [hunk 2]
```diff
@@ -55,6 +55,14 @@ func StringField(name string, value string) FieldOpt {
 	}
 }
 
+// UnicodeStringField adds a single UTF-16 string field to the event.
+func UnicodeStringField(name string, value string) FieldOpt {
+	return func(em *eventMetadata, ed *eventData) {
+		em.writeField(name, inTypeUnicodeString, outTypeString, 0)
```
Comment (kevpar) [inline pkg/etw/fieldopt.go:61]: Same question here about the out type.

Comment (pavelbazika) [reply pkg/etw/fieldopt.go:61]: Sorry for long delay, I missed the notification.

I followed the manifest here, which states
`<data name="Message" inType="win:UnicodeString" outType="xs:string"/>`

Unfortunately I don't have enough ETW technical background to know if `outTypeDefault` would suffice. The most I can do is to test if it works with `outTypeDefault`, but I can't be sure about some corner cases.

File: pkg/etw/fieldopt.go (hunk 1 above)
Comment (pavelbazika) [reply pkg/etw/fieldopt.go:80]: Please see below.

**Comment by helsaawy** [comment]
@pavelbazika
If you are still interested in merging this, could you rebase and push? it should be pretty quick to merge after that

File: pkg/etw/eventdata.go [hunk 3]
```diff
@@ -28,6 +29,13 @@ func (ed *eventData) writeString(data string) {
 	_ = ed.buffer.WriteByte(0)
 }
 
+// writeUnicodeString appends a string converted to UTF-16, including the null terminator, to the buffer.
+func (ed *eventData) writeUnicodeString(data string) {
+	unicode := utf16.Encode([]rune(data))
+	binary.Write(&ed.buffer, binary.LittleEndian, unicode)
+	ed.buffer.Write([]byte{0, 0})
```
Comment (helsaawy) [inline pkg/etw/eventdata.go:36]: to suppress the linter on error checking (or you can add a `//nolint:errcheck` directive)
```suggestion
	_ = binary.Write(&ed.buffer, binary.LittleEndian, unicode)
	_, _ = ed.buffer.Write([]byte{0, 0})
```

**Comment by kevpar** [comment]
Closing old PR. Please reactivate if this is still relevant.
